js_modules/dagster-ui/packages/ui-core/src/assets/BackfillPreviewModal.tsx

```diff
@@ -0,0 +1,172 @@
+import {gql, useQuery} from '@apollo/client';
+import {Box, Button, Colors, Dialog, DialogFooter, Spinner} from '@dagster-io/ui-components';
+import {useVirtualizer} from '@tanstack/react-virtual';
+import React from 'react';
+import styled from 'styled-components';
+
+import {tokenForAssetKey} from '../asset-graph/Utils';
+import {TargetPartitionsDisplay} from '../instance/backfill/TargetPartitionsDisplay';
+import {testId} from '../testing/testId';
+import {Container, HeaderCell, Inner, Row, RowCell} from '../ui/VirtualizedTable';
+
+import {AssetLink} from './AssetLink';
+import {asAssetKeyInput} from './asInput';
+import {AssetKey} from './types';
+import {
+  BackfillPreviewQuery,
+  BackfillPreviewQueryVariables,
+} from './types/BackfillPreviewModal.types';
+import {
+  BackfillPolicyForLaunchAssetFragment,
+  PartitionDefinitionForLaunchAssetFragment,
+} from './types/LaunchAssetExecutionButton.types';
+
+interface BackfillPreviewModalProps {
+  isOpen: boolean;
+  assets: {
+    assetKey: AssetKey;
+    partitionDefinition: PartitionDefinitionForLaunchAssetFragment | null;
+    backfillPolicy: BackfillPolicyForLaunchAssetFragment | null;
+  }[];
+  keysFiltered: string[];
+  setOpen: (isOpen: boolean) => void;
+}
+const TEMPLATE_COLUMNS = '1fr 1fr 1fr 1fr';
+
+export const BackfillPreviewModal = ({
+  isOpen,
+  setOpen,
+  assets,
+  keysFiltered,
+}: BackfillPreviewModalProps) => {
+  const assetKeys = React.useMemo(() => assets.map(asAssetKeyInput), [assets]);
+  const parentRef = React.useRef<HTMLDivElement | null>(null);
+
+  const rowVirtualizer = useVirtualizer({
+    count: assets.length,
+    getScrollElement: () => parentRef.current,
+    estimateSize: () => 60,
+    overscan: 10,
+  });
+  const totalHeight = rowVirtualizer.getTotalSize();
+  const items = rowVirtualizer.getVirtualItems();
+
+  const {data} = useQuery<BackfillPreviewQuery, BackfillPreviewQueryVariables>(
+    BACKFILL_PREVIEW_QUERY,
+    {
+      variables: {partitionNames: keysFiltered, assetKeys},
+      skip: !isOpen,
+    },
+  );
+
+  const partitionsByAssetToken = React.useMemo(() => {
+    return Object.fromEntries(
+      (data?.assetBackfillPreview || []).map((d) => [tokenForAssetKey(d.assetKey), d.partitions]),
+    );
+  }, [data]);
+
+  // BG Note: The transform: scale(1) below fixes a bug with MiddleTruncate where the text size
```

Comment:
ah weird that change `scale` doesnt trigger the resize observer

Reply:
I know right! This is still blowing my mind a bit, will try to come up with a better / more global fix in a follow-up